### ISSUE TITLE
Custom session data only persisted when set method is called on injected session

### ISSUE THREAD

**Issue by mdehoog** [simplabs/ember-simple-auth#805]
Related to #708.

I have a component into which I'm injecting the `session` service. I use the session to save some extra data that is persistent across reloads (an action saves the id of an entity in the session data):

``` javascript
export default Ember.Component.extend({
  session: Ember.inject.service('session'),
  actions: {
    manageBusiness: function (businessId) {
      this.get('session').set('data.businessId', businessId); //works fine
    }
  }
});
```

This works fine. However, if I change the `get.set` line to:

``` javascript
this.set('session.data.businessId', businessId);
```

...the data is no longer persistent across reloads. This is because the session service's `set` method (https://github.com/simplabs/ember-simple-auth/blob/2464beca521e0ee63c2833e4083cc3e405fbbb26/addon/services/session.js#L117) changes the `data.businessId` key to `session.businessId`, which causes the `businessId` property to be set on the `internal-session`. The `internal-session`'s `setUnknownProperty` method is then called, which updates the store (https://github.com/simplabs/ember-simple-auth/blob/2464beca521e0ee63c2833e4083cc3e405fbbb26/addon/internal-session.js#L102).

However, if you call `this.set('session.data.businessId', businessId);`, the session's `set` method is not hit and therefore the key is not replaced, and in turn the `internal-session`'s `setUnknownProperty` method is not called and the store isn't updated.

Note that it also works if I do the `data` -> `session` key replacement myself by calling `this.set('session.session.businessId', businessId);` (the `internal-session`'s `setUnknownProperty` is called).

**Comment by mdehoog** [comment]
Interestingly, because `session.data` is a `oneWay` computed property for `internal-session.content`, the value being set ends up in the same place. The only difference is that, when setting something in the internal-session's content object, `setUnknownProperty` isn't called which means `_updateStore` is not called.

**Comment by mdehoog** [comment]
Possible fix for this would be to change the session service property
```javascript
data: computed.oneWay('session.content')
```
to
```javascript
data: computed.oneWay('session')
```

**Comment by notmessenger** [comment]
I just ran into a variant of this.

```
this.get( 'session' ).set( 'data.flashData', 'on' );
this.get( 'session' ).set( 'data.user', record );
```

will persist the data across reloads but

```
this.get( 'session' ).setProperties({
    'data.flashData': 'on',
    'data.user': record
});
```

will not.

**Comment by gzurbach** [comment]
@mdehoog Thanks a lot for your detailed explanations, you probably saved me hours of investigation!

I switched to using `this.get('session').set('data.foo', 'bar');` for now and it seems to work perfectly.

**Comment by marcoow** [comment]
cleaning up old issues

**Comment by tehmaestro** [comment]
Hi, I ran into this issue and did a twiddle.
```
ember: 3.24.0
ember-simple-auth: 3.1.0
```

https://ember-twiddle.com/16647df0bdc19a9b199399b8a08d8023?openFiles=services.my-service%5C.js%2Ctemplates.components.my-component%5C.hbs

The first button and the third button are basically doing the same thing, but I figured I should try and make sure the value does not change. The second button uses the session service's own set, which works.
Note, I am grabbing the `_data` value from the ephemeral store, to show the lack of persistence, as local storage does not work on twiddle, which is what I am using in my project.
